spec/requests/api/article_published_status_spec.rb

```diff
@@ -0,0 +1,29 @@
+describe 'GET api/articles/:id', type: :request do
+  subject { response }
+
+  describe 'when the article is published' do
+    let!(:article) { create(:article) }
+    before do
+      get "/api/articles/#{article.id}"
+    end
+
+    it { is_expected.to have_http_status 200 }
+
+    it 'is expected to return an article with the published status true' do
+      expect(response_json['article']['published']).to eq true
+    end
+  end
+
+  describe 'when the article is not published' do
+    let!(:article) { create(:article, published: false)  }
+    before do
+      get "/api/articles/#{article.id}"
+    end
+
+    it { is_expected.to have_http_status 200 }
+
+    it 'is expected to return an article with the published status false' do
+      expect(response_json['article']['published']).to eq false
+    end
+  end
+end
```

Comment:
I don't see why you need separate test file for this. I think you can include asserting published status in already existing index or show spec

Reply:
Also I think if article is not published you probably don't want to see it returned by index action? I don't see not test nor functionality for that. If you planned that as separate feature I say you should include it here as just having published status is not a feature in of itself.